src/rpm/builder.rs

```diff
@@ -67,10 +67,27 @@ async fn async_file_mode(_file: &async_std::fs::File) -> Result<u32, RPMError> {
     Ok(0)
 }
 
+/// Compatibility profile
+///
+/// Older versions of RPM require certain feature flags to be enabled whereas
+/// newer versions assume their presence. We default to the `Legacy` profile for
+/// the widest possible compatibility.
+///
+/// TODO: this is internal-only (basically dead code) until the plan for "RPM v6"
+/// has been finalized upstream.
```

Comment:
@drahnr Options:

1) We can defer all of this until later, and I remove this code.  This PR gets pared down to just the removal of "Lead" from the API.
2) We can leave it like this, even though it's dead code, so that the some of the low-stakes changes can be worked on ahead of time
3) We can do #2, but via some different mechanism (like compile-time feature selection) instead of runtime configuration

Reply:
Depending how far the file formats align, different parsers while re-using data structures might be the best choice. Then hiding the different implementations behind a feature flag is secondary to that.

Reply:
I only skimmed the rpm V6 Format, but it doesn't appear to be too radically different. So the above outline approach should work.

Reply:
Since might be a while until rpm V6 hits the ground, and we don't want to be in the mess of not supporting existing installations when updating a tool that builds on-top of rpm-rs/rpm, we should be wary of the side effects and fallout. Even if it would be major server bump, we should always support more rather than less (or at least with a lot of lead time).

Reply:
Given the on-disk format isn't changing I don't think we need separate parsers.  Actually parsers in general don't seem problematic, it's more the writing of the RPMs that is.  We'd like to be able to produce "clean" modern RPMs without the legacy baggage, but we still need to support the baggage for users building e.g. CentOS 7 packages.